v3/spring-cloud-aws-v3-parameter-store/src/main/java/io/awspring/cloud/v3/paramstore/AwsParamStoreProperties.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.v3.paramstore;
+
+import java.net.URI;
+import java.util.regex.Pattern;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.util.StringUtils;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+/**
+ * Configuration properties for the AWS Parameter Store integration. Mostly based on the
+ * Spring Cloud Consul Configuration equivalent.
+ *
+ * @author Joris Kuipers
+ * @author Matej Nedic
+ * @since 2.0.0
+ */
+@ConfigurationProperties(AwsParamStoreProperties.CONFIG_PREFIX)
+public class AwsParamStoreProperties implements Validator {
```

Comment:
should this be moved to autoconfigure module? We can drop autoconfiguration dependency than in this module

Reply:
yes! I think the draft I shared already contains that change